server/src/main/java/org/opensearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -777,6 +780,179 @@ public void onFailure(Exception e) {
+            Map<String, BlobContainer> shardBlobs = indexBlob.getValue().children();
+            if (shardBlobs.isEmpty()) {
+                allIndicesListener.onResponse(null);
+            } else {
+                final ActionListener<List<Tuple<String, Tuple<Integer, RemoteStoreShardShallowCopySnapshot>>>> allShardsListener =
+                    new GroupedActionListener<>(allIndicesListener, shardBlobs.size());
+                for (Map.Entry<String, BlobContainer> shardBlob : shardBlobs.entrySet()) {
+                    executor.execute(new AbstractRunnable() {
+                        @Override
+                        protected void doRun() throws Exception {
+                            Map<String, BlobMetadata> shardLevelBlobs = shardBlob.getValue().listBlobs();
+                            List<Tuple<String, Tuple<Integer, RemoteStoreShardShallowCopySnapshot>>> resourcesToReleaseLock =
+                                new ArrayList<>();
+                            for (Map.Entry<String, BlobMetadata> shardLevelBlob : shardLevelBlobs.entrySet()) {
+                                String blob = shardLevelBlob.getKey();
+                                String snapshotUUID = blob.substring(SHALLOW_SNAPSHOT_PREFIX.length(), blob.length() - ".dat".length());
+                                if (blob.startsWith(SHALLOW_SNAPSHOT_PREFIX)
+                                    && blob.endsWith(".dat")
+                                    && (!allSnapshotIds.contains(snapshotUUID) || snapshotUUIDs.contains(snapshotUUID))) {
+                                    RemoteStoreShardShallowCopySnapshot remoteStoreShardShallowCopySnapshot =
+                                        REMOTE_STORE_SHARD_SHALLOW_COPY_SNAPSHOT_FORMAT.read(
+                                            shardBlob.getValue(),
+                                            snapshotUUID,
+                                            namedXContentRegistry
+                                        );
+                                    resourcesToReleaseLock.add(
+                                        Tuple.tuple(
+                                            snapshotUUID,
+                                            Tuple.tuple(Integer.valueOf(shardBlob.getKey()), remoteStoreShardShallowCopySnapshot)
+                                        )
+                                    );
+                                }
+                            }
+                            allShardsListener.onResponse(resourcesToReleaseLock);
+                        }
+
+                        @Override
+                        public void onFailure(Exception ex) {
+                            // Just passing null here to count down the listener instead of failing it, the stale data left behind
+                            // here will be retried in the next delete or repository cleanup
+                            allShardsListener.onResponse(null);
```

Comment:
we should lean towards failing the operation in these cases instead of suppressing it. It can leave inconsistent state if caller is moving with other operations and is unaware about failure in this operation.

Reply:
Outdated change no longer added